Tella/Data/Database/Vault/VaultDataSourceInterface.swift

```diff
@@ -0,0 +1,17 @@
+//
+//  Copyright © 2023 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+
+protocol VaultDataSourceInterface {
+    func addVaultFile(file : VaultFileDB, parentId: String?) throws
+    func getVaultFiles(parentId: String?, filter: FilterType?, sort: FileSortOptions?) -> [VaultFileDB]
```

Comment:
what do you think about using default parameter values when applicable.This simplifies method calls like this -> `parentId: String? = nil`

Reply:
Default argument is not permitted in a protocol method, but I can add it in an extension.
I don't think I have a case where I assign nil to the `parentId` parameter